src/test/java/io/github/jhipster/online/service/JdlServiceTest.java

```diff
@@ -0,0 +1,196 @@
+        jdlService =
+            new JdlService(logsService, gitService, jHipsterService, githubService, gitlabService, jdlRepository, applicationProperties);
+    }
+
+    @Test
+    void applyJdlWithException() {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+
+        jdlService.applyJdl(user, "jhipster", "online", jdlMetadata, applyJdlId, GitProvider.GITHUB);
+
+        verify(logsService).addLog(applyJdlId, "Error during generation: Error creating file jhipster-jdl.jh, could not write the file");
+        verify(logsService).addLog(applyJdlId, "Generation failed");
+    }
+
+    @Test
+    void applyGithubJdl() throws GitAPIException, IOException {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+        jdlMetadata.setId("1");
+        Jdl jdl = mock(Jdl.class);
+        jdlMetadata.setName("jdl-metadata-name");
+        given(jdlRepository.findOneByJdlMetadataId(jdlMetadata.getId())).willReturn(Optional.of(jdl));
+        given(githubService.getHost()).willReturn("https://www.github.com");
+
+        jdlService.applyJdl(user, "jhipster", "online", jdlMetadata, applyJdlId, GitProvider.GITHUB);
+
+        verifyApplyJdl(applyJdlId, jdlMetadata);
+        verify(logsService).addLog(applyJdlId, "Cloning GitHub repository `jhipster/online`");
+        verify(gitService).cloneRepository(eq(user), any(File.class), eq("jhipster"), eq("online"), eq(GitProvider.GITHUB));
+        verify(logsService).addLog(applyJdlId, "Pushing the application to the GitHub remote repository");
+        verify(gitService).push(nullable(Git.class), any(File.class), eq(user), eq("jhipster"), eq("online"), eq(GitProvider.GITHUB));
+        verify(logsService).addLog(applyJdlId, "Creating Pull Request");
+        verify(githubService)
+            .createPullRequest(
+                user,
+                "jhipster",
+                "online",
+                "Add entities using the JDL model `jdl-metadata-name`",
+                "jhipster-entities-apply-id",
+                "Entities generated by JHipster using the model at https://start.jhipster.tech/jdl-studio/#!/view/1"
+            );
+        verify(logsService).addLog(applyJdlId, "Pull Request created at https://www.github.com/jhipster/online/pull/0");
+    }
+
+    @Test
+    void applyGitlabJdl() throws GitAPIException, IOException {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+        jdlMetadata.setId("1");
+        Jdl jdl = mock(Jdl.class);
+        jdlMetadata.setName("jdl-metadata-name");
+        given(jdlRepository.findOneByJdlMetadataId(jdlMetadata.getId())).willReturn(Optional.of(jdl));
+        given(gitlabService.getHost()).willReturn("https://www.gitlab.com");
```

Comment:
I would use the property `application.gitlab.host` from `application.yml` instead of hardcoding `https://www.gitlab.com`